consumer.go

```diff
@@ -126,7 +126,7 @@ type Consumer struct {
 
 	// used at connection close to force a possible reconnect
 	lookupdRecheckChan chan int
-	lookupdHTTPAddrs   []string
+	lookupdHTTPAddrs   []lookupdHTTPAddr
```

Comment:
I don't see any benefit of introducing a type here, let's keep this a `string` and then a few of the other changes here won't be necessary.

Reply:
Ok, I removed the type.